### PR TITLE
[#5] Write some README

### LINKED ISSUE #5: Add README documentation

### PR DESCRIPTION
**Pull request by vrom911**
Resolves #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,64 @@
 [![Stackage Nightly](http://stackage.org/package/postgresql-simple-named/badge/nightly)](http://stackage.org/nightly/package/postgresql-simple-named)
 [![Build status](https://img.shields.io/travis/Holmusk/postgresql-simple-named.svg?logo=travis)](https://travis-ci.org/Holmusk/postgresql-simple-named)
```
Comment (chshersh) [inline README.md]: Should we rearrange badges for consistency?

File: src/PgNamed.hs [hunk 2]
```diff
@@ -4,8 +4,9 @@
...
 {- | Introduces named parameters for @postgresql-simple@ library.
 It uses @?@ question mark symbol as the indicator of the named parameter which
-is replaced with the standard syntax with question marks. Check out the example
-of usage:
+is replaced with the standard syntax with question marks.
+
+Check out the example of usage:
 
 @
 queryNamed [sql|
```
Comment (chshersh) [inline src/PgNamed.hs]: Let's make it also hyperlink
```suggestion
'queryNamed' [sql|
```

File: src/PgNamed.hs [hunk 3]
```diff
@@ -28,6 +29,7 @@ module PgNamed
 
          -- * Errors
        , PgNamedError (..)
+       , WithError
```
Comment (chshersh) [inline src/PgNamed.hs]: If we exporting this type alias, do you think it makes sense to rename it to `WithNamedError` to not be too generic? I doubt that somebody will use this constraint, though (we don't use, we have our own error type). So it's merely for the documentation

File: README.md [hunk 4]
```diff
@@ -6,7 +6,64 @@
...
+    FROM my_table
+    WHERE id = ?
+      AND (? IS NULL OR id > ? )
+      AND (? IS NULL OR id < ? )
+
+|] (someId, minId, minId, maxId, maxId)
+```
+
+This is how the query looks like with the `postgresql-simple` library. You can
+rewrite it the following way:
```
Comment (chshersh) [inline README.md]:
```suggestion
rewrite it the following way using the `postgresql-simple-named` library:
```

File: README.md [hunk 5]
```diff
@@ -6,7 +6,64 @@
 [![Stackage Nightly](http://stackage.org/package/postgresql-simple-named/badge/nightly)](http://stackage.org/nightly/package/postgresql-simple-named)
 [![Build status](https://img.shields.io/travis/Holmusk/postgresql-simple-named.svg?logo=travis)](https://travis-ci.org/Holmusk/postgresql-simple-named)
 
-Implementation of named parameters for `postgresql-simple` library
+This library introduces the implementation of named parameters for the
+[`postgresql-simple`][pgs] library. `postgresql-simple-named` is considered to
+be used along with the [`postgresql-simple`][pgs] library, so you could refer
+there for the original documentation of primary functions. This package solves
+exclusively one particular problem — gives the ability to use named parameters
+instead of `?` in quasi-quoter queries and offers essential functions for using
```
Comment (chshersh) [inline README.md]: Jus to use `use` not so often 🙂 
```suggestion
instead of `?` in quasi-quoter queries and offers essential functions for substituting variables
```
(and delete `them` on next line)

File: README.md [hunk 6]
```diff
@@ -6,7 +6,64 @@
...
+them in queries (`queryNamed`, `executeNamed`).
+
+## Example
+
+Using the provided operator `=?` for named parameters creation and using '?'
+prefix for specifying the parameters inside the queries, here is how the query
+could look like:
```
Comment (chshersh) [inline README.md]: I would rephrase this like this:

```
Operator `=?` binds named parameters with the corresponding values. 
Named parameters inside SQL query start with the '?' character and can contain lowercase
and uppercase letters, digits and underscore. Below you can find a basic example of how
query with named parameters could look like:
```

File: README.md [hunk 7]
```diff
@@ -6,7 +6,64 @@
...
+    FROM users
+    WHERE name = ?nameParam
+      AND age  = ?ageParam
+
+|] [ "nameParam" =? "John"
+   , "ageParam"  =? 42
+   ]
+```
+
+This feature can be extremely helpful when the query using some parameters more than once:
```
Comment (chshersh) [inline README.md]:
```suggestion
This feature can be extremely helpful when the query uses some parameters more than once:
```

**Review by chshersh — COMMENTED** [review]
Documentation is great! I think we shouldn't write an essay here, the current description shows the problem and solution quite lovely 🙂

File: src/PgNamed.hs (hunk 3 above)
Comment (vrom911) [reply src/PgNamed.hs]: Yes, I was thinking about it as well. Good point.
The reason I do export it is because the documentation for query functions won't have mystical `WithError`. Will do the renaming!

**Review by chshersh — APPROVED** [review]
This looks great!
